.github/workflows/shared-create-hotfix-pr.yml

```diff
@@ -0,0 +1,180 @@
+name: Shared / Create Hotfix PR
+
+# This workflow creates a Pull Request towards support branches. 
+# A changelog, including the new version, is generated and version strings in relevant files are replaced. 
+# All these changes are committed and submitted as a Pull Request.
+
+on:
+  workflow_call:
+    inputs:
+      tag-prefix:
+        description: "will be pre-pended to tag-base"
+        required: false
+        type: string
+      branch:
+        description: "branch to hotfix on"
+        required: true
+        type: string
+      main-tag-regex:
+        description: "the regex to find all related main releases"
+        required: true
+        type: string
+      changelog-path:
+        description: "path to the changelog file"
+        required: false
+        default: ./CHANGELOG.md
+        type: string
+      changelog-config-path:
+        description: "path to the changelog config"
+        required: true
+        type: string
+      pr-body-text:
+        description: "text to be included in the PR"
+        required: false
+        type: string
+      release-target:
+        description: "target of the release (rust|wasm)"
+        required: true
+        type: string
+    secrets:
+      GPG_PRIVATE_KEY:
+        description: "GPG private key for signing commits and tags"
+        required: true
+      GPG_PASSPHRASE:
+        description: "GPG private passphrase for signing commits and tags"
+        required: true
+        
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
+          fetch-depth: 0
+          ref: ${{ inputs.branch }}
+
+      - name: Determine Hotfix Version
+        run: |
+
+          TAG_PREFIX=${{ inputs.tag-prefix }}
+          BRANCHNAME=${{ inputs.branch }}
+
+          SUPPORT_TAG_BASE=${BRANCHNAME##*/}
+
```

Comment:
Does this do any validation that the support tag is correct?
What happens when I create and pass in an invalid branch e.g. `release/v1.2` instead of `support/v1.2`, or `support/v1.2.3`, or `support/v1`, or `support/wasm-v1.2` on the Rust workflow or vice-versa?

Reply:
how about the pass in a regex for the shared workflow. Then the implementing workflow decides what is valid input.